packages/block-library/src/navigation/index.php

```diff
@@ -144,12 +144,14 @@ function block_core_navigation_add_directives_to_submenu( $w, $block_attributes
 
 	/**
 	 * Replaces view script for the Navigation block with version using Interactivity API.
+	 * //phpcs:disable Gutenberg.NamingConventions.ValidBlockLibraryFunctionName
 	 *
 	 * @param array $metadata Block metadata as read in via block.json.
 	 *
 	 * @return array Filtered block type metadata.
 	 */
 	function gutenberg_block_core_navigation_update_interactive_view_script( $metadata ) {
```

Comment:
> Aside, we should also enforce that you can't use functions prefixed with gutenberg_ with core blocks in the same set of files as they don't exist in WP core.

Sure. This can enforced. I will update the PR, @gziolo.

Reply:
> @luisherranz, are you aware of this issue?

Not at all. I don't have any experience backporting stuff to WP core yet, nor with how it affects code here in Gutenberg. Happy to fix things if you give me instructions on what needs to be done, tho.

cc: @westonruter (pinging because Weston introduced the `gutenberg_should_block_use_interactivity_api`` function, just in case)

Reply:
Related to that, I think it could be valuable to include a mandatory banner in every PHP file we know gets backported to WordPress core with the link to the document explaining all the expected requirements enforced by the complex process. It will also give a hint to folks whenever they try to update those files directly in the WordPress core which is going to be replaced the next time someone runs `npm run build` ...

Reply:
> Sure. This can enforced. I will update the PR, @gziolo.

I didn't realize that the linting rule doesn't allow any other functions to be defined outside of the ones specified in the `prefixes` config property.
Therefore, there's no need to modify the PR. 🤦‍♂️

The linting rule will only permit functions that start with the `block_core_`, `render_block_core_`, and `register_block_core_ prefixes` (+ the directory name).

The two functions with the `gutenberg_` prefix (`gutenberg_block_core_file_update_interactive_view_script` and `gutenberg_block_core_navigation_update_interactive_view_script`) were allowed because of the `phpcs::disable` statements before them.

Given the discussion above, I'm uncertain about the course of action for these two `gutenberg_` prefixed functions. Should they be allowed for now?

cc @gziolo @luisherranz